src/Hyperparameters.jl

```diff
@@ -109,28 +116,38 @@ end
 Save value to the enviroment variables and the global `HYPERPARAMETERS` dictionary.
 """
 function save_hyperparam(name::Symbol, value; prefix::AbstractString="")
-    ENV[uppercase(string(prefix, name))] = string(value)
+    ENV[_name_to_envvar(prefix, name)] = string(value)
     _set_hyperparam(name, value)
 end
 
 """
-    report_hyperparameters(save_dir::AbstractPath)
+    report_hyperparameters(save_dir::AbstractPath; prefix="$SAGEMAKER_PREFIX")
 
-Saves the cached `HYPERPARAMETERS` to a JSON file named "hyperparameters.json" in the `save_dir`
+Saves all hyperparameters to a JSON file named "hyperparameters.json" in the `save_dir`
 and prints each key-value pair to the logger.
 
+The hyperparameters are taken from the cached `HYPERPARAMETERS` dictionary of all that were
+used, combined with any enviroment variables matching the prefix.
```

Comment:
What happens if `save_hyperparam` is used on a worker? Should we add a note to that docstring if reporting may not work?

Reply:
I don't know. We don't use that function anywhere anymore.
Might remove it in 2.0?